aclk/aclk_otp.c

```diff
@@ -248,6 +251,7 @@ void aclk_get_mqtt_otp(RSA *p_key, char *aclk_hostname, int port, char **mqtt_us
     debug(D_ACLK, "Password phase: %s",response_json);
 
     https_req_response_free(&resp);
+    memset(&resp, 0, sizeof(https_req_response_t));
```

Comment:
That defeats the whole purpose of the HTTPS_RES_RESPONSE_T_INITIALIZER macro, doesn't it? Thinking it's better to add a proper initialization function.

Reply:
@ 198 you use it to initialize without calling any function

Reply:
I added init function and retained initializer... same idea is used by e.g. pthreads API